src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs

```diff
@@ -3,56 +3,94 @@
         {
-            _podName = environment.GetEnvironmentVariable(EnvironmentSettingNames.PodName);
             _writeEvent = writeEvent ?? ConsoleWriter;
         }
 
         public override void LogFunctionTraceEvent(LogLevel level, string subscriptionId, string appName, string functionName, string eventName, string source, string details, string summary, string exceptionType, string exceptionMessage, string functionInvocationId, string hostInstanceId, string activityId, string runtimeSiteName, string slotName, DateTime eventTimestamp)
         {
-            string formattedEventTimeStamp = eventTimestamp.ToString(EventTimestampFormat);
-            string hostVersion = ScriptHost.Version;
             FunctionsSystemLogsEventSource.Instance.SetActivityId(activityId);
             details = details.Length > MaxDetailsLength ? details.Substring(0, MaxDetailsLength) : details;
 
-            _writeEvent($"{(int)ToEventLevel(level)},{subscriptionId},{appName},{functionName},{eventName},{source},{NormalizeString(details)},{NormalizeString(summary)},{hostVersion},{formattedEventTimeStamp},{exceptionType},{NormalizeString(exceptionMessage)},{functionInvocationId},{hostInstanceId},{activityId},{runtimeSiteName},{slotName},{_podName}");
+            JObject traceLog = new JObject();
+            traceLog.Add("EventType", ScriptConstants.LinuxLogEventStreamName);
+            traceLog.Add("Level", (int)ToEventLevel(level));
+            traceLog.Add("SubscriptionId", subscriptionId);
+            traceLog.Add("AppName", appName);
+            traceLog.Add("FunctionName", functionName);
+            traceLog.Add("EventName", eventName);
+            traceLog.Add("Source", source);
+            traceLog.Add("Details", NormalizeString(details));
+            traceLog.Add("Summary", NormalizeString(summary));
+            traceLog.Add("HostVersion", ScriptHost.Version);
+            traceLog.Add("EventTimeStamp", eventTimestamp.ToString(EventTimestampFormat));
+            traceLog.Add("ExceptionType", exceptionType);
+            traceLog.Add("ExceptionMessage", NormalizeString(exceptionMessage));
+            traceLog.Add("FunctionInvocationId", functionInvocationId);
+            traceLog.Add("HostInstanceId", hostInstanceId);
+            traceLog.Add("ActivityId", activityId);
+            traceLog.Add("RuntimeSiteName", runtimeSiteName);
+            traceLog.Add("SlotName", slotName);
+
+            _writeEvent(traceLog.ToString(Formatting.None));
         }
 
         public override void LogFunctionMetricEvent(string subscriptionId, string appName, string functionName, string eventName, long average, long minimum, long maximum, long count, DateTime eventTimestamp, string data, string runtimeSiteName, string slotName)
         {
-            string hostVersion = ScriptHost.Version;
+            JObject metricEvent = new JObject();
+            metricEvent.Add("EventType", ScriptConstants.LinuxLogEventStreamName);
```

Comment:
Please add comments - how these prefixes will used when host is running in Kubenetes env.